tests/demo/latching_topic_test.py

```diff
@@ -0,0 +1,22 @@
+# QoS Profile
+from rclpy.qos import DurabilityPolicy, HistoryPolicy, QoSProfile
+
+# ROS2 interfaces
+from std_msgs.msg import String
+
+# Testing
+from ros2_easy_test import ROS2TestEnvironment, with_single_node
+
+# Module under test and interfaces
+from ..example_nodes.well_behaved import LatchingNode
+
+
+@with_single_node(
+    LatchingNode,
+    watch_topics={"/hello": String},
+    qos_profiles={
+        "/hello": QoSProfile(durability=DurabilityPolicy.TRANSIENT_LOCAL, history=HistoryPolicy.KEEP_ALL)
```

Comment:
I thought of that too. But most people probably won't specify a QoS Profile or only specify watch_topics for a topic and qos_profiles for another one. If we change the watch_topics syntax, people would have to refactor their calls to explicitly set the profile to None. We could that via some if statements but in my opinion it gets quite ugly. that way

Reply:
This can be made conditional by allow only a `Type` or a `(Type, QoS)` tuple? That would keep backwards compatibility.

But it might be confusing, so I'm good with either one.

Reply:
I think the `(Type, QoS)` is very clean, too.

BUT: That way, we don't have a nice way of specifying it for publishers. Thus, I think we should keep it as-is.

Reply:
For publishers I would simply add an extra argument to the `.publish` function.

Reply:
I don't think this is a good idea since it only affects the very first time, but the method suggests otherwise. Also, if we don't repeat it, reordering messages can have unexpected side effects.

Reply:
Then let's leave the proposal as it is!